### PR TITLE
fix: stack bar transparent and refactor stacked bar

### PR DESCRIPTION
**Pull request by rokamu623**
## Description
In graphs generated by `Plot.create_response_time_stacked_bar_plot()` the ability to make the drawing of certain callbacks semi-transparent is not working correctly.

Refactoring of `caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py` was done in association with the resolution of this issue.

## Related links
Jira: https://tier4.atlassian.net/browse/RT2-806

Test scripit: [Jira->説明->path.zip](https://tier4.atlassian.net/browse/RT2-806)

Github(see also):
https://github.com/tier4/CARET_analyze/pull/265/files
- Implementation of stacked bar plot

https://github.com/tier4/CARET_analyze/pull/287/files
- Implementation of separate hover for each bar.
- However, a transparency issue occurred here.

## Notes for reviewers
- Does hover move correctly?
- Does the click policy (transparency) work correctly?

regression test of CARET_report
image

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py [hunk 1]
```diff
@@ -80,91 +78,33 @@ def create_figure(self) -> Figure:
...
-            raise NotImplementedError()
-        else:  # index
-            output_data[x_label] = list(range(0, len(output_data[y_labels[0]])))
-            x_width_list = BokehStackedBar._get_x_width_list(output_data[x_label])
-
-        return output_data, x_width_list
-
-    @staticmethod
-    def _updated_with_unit(
+    def updated_with_unit(
```
Comment (atsushi421) [inline src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:107]: Move this function to the StackedBarSource class.

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py [hunk 2]
```diff
@@ -201,7 +132,7 @@ def _updated_timestamps_to_offset_time(
         return new_values
 
     @staticmethod
-    def _get_x_width_list(x_values: list[float]) -> list[float]:
+    def get_x_width_list(x_values: list[float]) -> list[float]:
```
Comment (atsushi421) [inline src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:135]: Move this function to the StackedBarSource class.

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py [hunk 3]
```diff
@@ -223,7 +154,7 @@ def _get_x_width_list(x_values: list[float]) -> list[float]:
         return x_width_list
 
     @staticmethod
-    def _add_shift_value(
+    def add_shift_value(
```
Comment (atsushi421) [inline src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:157]: Move this function to the StackedBarSource class.

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py [hunk 4]
```diff
@@ -80,91 +78,33 @@ def create_figure(self) -> Figure:
...
-                source=stacked_bar_source.generate(y_label, stacked_bar_data, x_width_list),
-                color=color_selector.get_color(y_label)
-            )
-            vbar_stacks.append((y_label, vbar_stack))
-
-        legend = Legend(items=vbar_stacks, location='bottom_left')
+        colors = [color_selector.get_color(y_label) for y_label in y_labels]
+
+        source = StackedBarSource(data, y_labels, self._xaxis_type, x_label)
+        # stack bottom -> top to top -> bottom
```
Comment (atsushi421) [inline src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:84]: This comment needs a little more clarification.

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py [hunk 5]
```diff
@@ -80,91 +78,33 @@ def create_figure(self) -> Figure:
...
+        stacks = fig.vbar_stack(reversed_y_label, x='start time', width='x_width_list',
+                                color=colors, source=source.to_source())
+
+        # add 'label' and 'latency' data to each bar due to display hover
+        for label, stack in zip(reversed_y_label, stacks):
+            stack.data_source.add([label] * source.x_len, 'label')
+            stack.data_source.add(['latency = ' + str(latency)
+                                   for latency in source.data[label]], 'latency')
+            fig.add_tools(
+                HoverKeysFactory.create_instance('stacked_bar', target_objects).create_hover())
```
Comment (atsushi421) [inline src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:95]: It is sufficient to call `add_tools` once.
If there is no particular reason, this should be done using the StackedBarSource.create_hover(), as in the previous implementation.

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py [hunk 6]
```diff
@@ -80,91 +78,33 @@ def create_figure(self) -> Figure:
...
+        # stack bottom -> top to top -> bottom
+        reversed_y_label = list(reversed(y_labels))
+        stacks = fig.vbar_stack(reversed_y_label, x='start time', width='x_width_list',
+                                color=colors, source=source.to_source())
+
+        # add 'label' and 'latency' data to each bar due to display hover
+        for label, stack in zip(reversed_y_label, stacks):
+            stack.data_source.add([label] * source.x_len, 'label')
+            stack.data_source.add(['latency = ' + str(latency)
+                                   for latency in source.data[label]], 'latency')
```
Comment (atsushi421) [inline src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:93]: Moving this process to StackedBarSource.to_source() improves computational complexity and module coupling.

**Review by atsushi421 — CHANGES_REQUESTED** [review]
@rokamu623 
It would be great if StackedBarSource.to_source() could be modified to return the ColumnDataSource so that it is consistent with other drawing methods.

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py (hunk 1 above)
Comment (rokamu623) [reply src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:155]: Fixed it in [7413d0f](https://github.com/tier4/CARET_analyze/pull/307/commits/7413d0f72a3ce8e0052c712da4b7db406081e5be).

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py (hunk 2 above)
Comment (rokamu623) [reply src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:173]: Fixed it in [7413d0f](https://github.com/tier4/CARET_analyze/pull/307/commits/7413d0f72a3ce8e0052c712da4b7db406081e5be).

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py (hunk 3 above)
Comment (rokamu623) [reply src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py:195]: Fixed it in [7413d0f](https://github.com/tier4/CARET_analyze/pull/307/commits/7413d0f72a3ce8e0052c712da4b7db406081e5be).

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py (hunk 4 above)
Comment (rokamu623) [reply src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py]: Fixed it in [dbdc031](https://github.com/tier4/CARET_analyze/pull/307/commits/dbdc0311e5097a031aa934d83f464cacba3d933b).

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py (hunk 5 above)
Comment (rokamu623) [reply src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py]: Fixed it in [dbdc031](https://github.com/tier4/CARET_analyze/pull/307/commits/dbdc0311e5097a031aa934d83f464cacba3d933b).

File: src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py (hunk 6 above)
Comment (rokamu623) [reply src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py]: Fixed it in [7d58270](https://github.com/tier4/CARET_analyze/pull/307/commits/7d582702f850e8001e84b9cb31345618c675ba6f).

**Comment by rokamu623** [comment]
@atsushi421 
Please reconfirm.

@emori-ctc 
Please review.

**Review by atsushi421 — APPROVED** [review]
LGTM

**Comment by rokamu623** [comment]
@takeshi-iwanari 
please review.

**Review by takeshi-iwanari — APPROVED** [review]
LGTM

**Pull request merged**